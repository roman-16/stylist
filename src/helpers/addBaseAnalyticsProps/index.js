import { mapKeys } from 'lodash-es';
import context from '@context';

export default ({ className, category, ...data }) => ({
  ...mapKeys(data, (value, key) => `data-${key}`),
  className: className ?? 'track-click',
  'data-category': category ?? context.pageType,
});
