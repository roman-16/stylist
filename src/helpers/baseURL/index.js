import context from '@context';
import env from '@env';

const baseURL = () => !!env.BASE_URL;

baseURL.getPathname = (pathname, addLocale = false) => {
  if (!baseURL()) return pathname;

  const url = new URL(env.BASE_URL, window.location.origin);

  url.pathname = pathname;

  if (addLocale) {
    url.pathname = `/${context.language}-${context.country}${url.pathname}`;
  }

  return url.href;
};

export default baseURL;
