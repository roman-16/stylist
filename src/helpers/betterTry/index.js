export default (func) => {
  let result;
  let error;

  try {
    result = func();
  } catch (_error) {
    error = _error;
  }

  if (result?.then) {
    return result.then((_result) => [_result]).catch((_error) => [undefined, _error]);
  }

  return [result, error];
};
