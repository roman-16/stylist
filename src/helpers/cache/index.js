const prefix = 'bt-widgets-cache-';
const getSeconds = () => Math.round(Date.now() / 1000);

const cache = (key) => {
  const cacheKey = prefix + key;
  const cacheHelper = {
    remove: () => localStorage.removeItem(cacheKey),
  };

  return {
    ...cacheHelper,

    set: (value, ttl = 60) => {
      const futureTTL = getSeconds() + ttl;

      localStorage.setItem(
        cacheKey,
        JSON.stringify({
          ttl: futureTTL,
          value,
        }),
      );
    },

    get: () => {
      const data = JSON.parse(localStorage.getItem(cacheKey));

      if (data === null || data.ttl <= getSeconds()) {
        cacheHelper.remove();

        return undefined;
      }

      return data.value;
    },
  };
};

cache.clear = () => {
  Object.keys(localStorage).forEach((key) => {
    if (key.startsWith(prefix)) {
      localStorage.removeItem(key);
    }
  });
};

export default cache;
