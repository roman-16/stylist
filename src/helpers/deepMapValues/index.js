import { mapValues, isPlainObject } from 'lodash-es';

const deepMapValues = (object, callback, { scope = '' } = {}) =>
  mapValues(object, (value, key) => {
    if (isPlainObject(value)) {
      return deepMapValues(value, callback, {
        scope: scope ? `${scope}.${key}` : key,
      });
    }

    return callback(value, key, { scope });
  });

export default deepMapValues;
