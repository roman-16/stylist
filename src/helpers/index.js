export { default as addBaseAnalyticsProps } from './addBaseAnalyticsProps';
export { default as baseURL } from './baseURL';
export { default as betterTry } from './betterTry';
export { default as cache } from './cache';
export { default as deepMapValues } from './deepMapValues';
export { default as randomId } from './randomId';
export { default as removeLocalFromPath } from './removeLocalFromPath';
export { default as resolveTemplate } from './resolveTemplate';
export { default as scene7 } from './scene7';
export { default as searchParams } from './searchParams';
