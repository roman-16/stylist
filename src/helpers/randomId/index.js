let counter = 0;
const random = () => Math.floor(Math.random() * 1000000);
const randomId = () => `r-${counter++}-${random()}`;

randomId.number = () => random() * 1000 + counter++;

export default randomId;
