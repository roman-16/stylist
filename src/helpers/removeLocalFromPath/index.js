export default (url) => url.split('/').slice(2).join('/');
