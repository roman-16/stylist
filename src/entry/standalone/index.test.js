import { attachDiv } from '@@jest/helpers';
// import { act } from '@testing-library/react';
import '.';

attachDiv.onEach('app');

it('sets a window object', () => {
  expect(window.bt).toMatchSnapshot();
});

// FIXME: THIS TEST DOESN'T WORK AND I DON'T KNOW WHY
// eslint-disable-next-line jest/no-commented-out-tests
// it('renders a widget', async () => {
//   let id = '';

//   await act(async () => {
//     id = await window.bt.widgets.renderBlueWorldStories({
//       selector: '#app',
//       position: 'after',
//     });
//   });

//   expect(document.getElementById(id)).toBeInTheDocument();
// });
