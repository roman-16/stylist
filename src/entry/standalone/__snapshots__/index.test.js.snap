// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets a window object 1`] = `
Object {
  "widgets": Object {
    "BuyersGuide": Object {
      "renderCalculator": [Function],
      "renderProductRecommendation": [Function],
      "renderProductSwiper": [Function],
      "renderSnowboardFinder": [Function],
    },
    "ShopYourStyle": Object {
      "Explorer": Object {
        "renderStandalone": [Function],
        "renderWithData": [Function],
      },
      "renderMoreStyles": [Function],
      "renderProductsOverview": [Function],
      "renderStyleLike": [Function],
    },
    "renderBlueWorldSearch": [Function],
    "renderBlueWorldStories": [Function],
    "renderCheckoutFeedbackForm": [Function],
    "renderPDPFeedbackForm": [Function],
    "renderSnowboardSwap": [Function],
  },
}
`;
